File: VB/WebSite/App_Code/CustomResource.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Web
#Region "#usingresources"
Imports System.Collections
#End Region ' #usingresources

#Region "#customresources"
<Serializable> _
Public Class CustomResource
	Private resourceID_Renamed As Object
	Private name_Renamed As String

	Public Sub New()
	End Sub

	Public Sub New(ByVal ID As Object, ByVal nameValue As String)
		Me.resourceID_Renamed = ID
		Me.name_Renamed = nameValue
	End Sub


	Public Property Name() As String
		Get
			Return name_Renamed
		End Get
		Set(ByVal value As String)
			name_Renamed = value
		End Set
	End Property
	Public Property ResourceID() As Object
		Get
			Return resourceID_Renamed
		End Get
		Set(ByVal value As Object)
			resourceID_Renamed = value
		End Set
	End Property
End Class

<Serializable> _
Public Class CustomResourceList
	Inherits List(Of CustomResource)
End Class

Public Class CustomResourceDataSource
	Private events As CustomResourceList
	Public Sub New(ByVal events As CustomResourceList)
		If events Is Nothing Then
			DevExpress.XtraScheduler.Native.Exceptions.ThrowArgumentNullException("resources")
		End If
		Me.events = events
	End Sub
	Public Sub New()
		Me.New(New CustomResourceList())
	End Sub
	Public Property Resources() As CustomResourceList
		Get
			Return events
		End Get
		Set(ByVal value As CustomResourceList)
			events = value
		End Set
	End Property
	Public ReadOnly Property Count() As Integer
		Get
			Return Resources.Count
		End Get
	End Property

	Public Function SelectMethodHandler() As IEnumerable
		Dim result As New CustomResourceList()
		result.AddRange(Resources)
		Return result
	End Function
End Class
#End Region ' #customresources

File: CS/WebSite/App_Code/CustomResource.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
#region #usingresources
using System.Collections;
#endregion #usingresources

#region #customresources
[Serializable]
public class CustomResource
{
    object resourceID;
    string name;

    public CustomResource()
    {
    }

    public CustomResource(object ID, string nameValue)
    {
        this.resourceID = ID;
        this.name = nameValue;
    }


    public string Name { get { return name; } set { name = value; } }
    public object ResourceID { get { return resourceID; } set { resourceID = value; } }
}

[Serializable]
public class CustomResourceList : List<CustomResource>
{
}

public class CustomResourceDataSource
{
    CustomResourceList events;
    public CustomResourceDataSource(CustomResourceList events)
    {
        if (events == null)
            DevExpress.XtraScheduler.Native.Exceptions.ThrowArgumentNullException("resources");
        this.events = events;
    }
    public CustomResourceDataSource()
        : this(new CustomResourceList())
    {
    }
    public CustomResourceList Resources { get { return events; } set { events = value; } }
    public int Count { get { return Resources.Count; } }

    public IEnumerable SelectMethodHandler()
    {
        CustomResourceList result = new CustomResourceList();
        result.AddRange(Resources);
        return result;
    }
}
#endregion #customresources